lib/types/semantic-errors/BranchError.ts

```diff
@@ -0,0 +1,12 @@
+export class BranchError extends Error {
```

Comment:
Could this just be `types/errors/...` to be shorter?

Reply:
I think we should move this to `lib/util/errors` as those are used classes and not type only.

`lib/types` was planned for typescript only types, with are removed on runtime.